### PR TITLE
8338426: Test java/nio/channels/Selector/WakeupNow.java failed

### PR DESCRIPTION
**Pull request by bplb**
`System.currentTimeMillis()` is replaced with `System.nanoTime()` and the elapsed time is added to the failure exception message.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8338426](https://bugs.openjdk.org/browse/JDK-8338426): Test java/nio/channels/Selector/WakeupNow.java failed (**Bug** - P4)

### Reviewers
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21674/head:pull/21674` \
`$ git checkout pull/21674`

Update a local copy of the PR: \
`$ git checkout pull/21674` \
`$ git pull https://git.openjdk.org/jdk.git pull/21674/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21674`

View PR using the GUI difftool: \
`$ git pr show -t 21674`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21674.diff">https://git.openjdk.org/jdk/pull/21674.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21674#issuecomment-2433991978)

### CODE REVIEW

**Comment by bplb** [comment]
The validity of using a temporal threshold on elapsed time as a failure criterion, as well as the value of that threshold, are not considered at this time.

File: test/jdk/java/nio/channels/Selector/WakeupNow.java [hunk 1]
```diff
@@ -47,33 +49,33 @@ private static void test1() throws Exception {
...
+            throw new RuntimeException("test failed with delta " + delta);
     }
 
     // Test if selectNow clears wakeup with only the wakeup fd
     // in the selector.
     // This fails before the fix on Solaris
     private static void test2() throws Exception {
         Selector sel = Selector.open();
-        Pipe p = Pipe.open();
-        p.source().configureBlocking(false);
```
Comment (jaikiran) [inline test/jdk/java/nio/channels/Selector/WakeupNow.java:66]: Hello Brian, is this intentionally being removed?

Comment (jaikiran) [reply test/jdk/java/nio/channels/Selector/WakeupNow.java:66]: Looking at this test, this removal appears intentional and it's merely cleaning up unused (and thus irrelevant) `Pipe`. It appears that this was a copy/paste oversight in the original test (even the comment on `test2()` method seems to indicate this `Pipe` was an oversight). So what you did here looks correct.

**Review by jaikiran — APPROVED** [review]
This change to move away from using `System.currentTimeMillis()` in favour of `System.nanoTime()` looks OK to me.

**Comment by AlanBateman** [comment]
Changing it to use nanoTime is good but I wonder about the reliability of the select(2000) taking >= 1s. This test has a history of being on and off the exclude list several times over the years.

File: test/jdk/java/nio/channels/Selector/WakeupNow.java (hunk 1 above)
Comment (bplb) [reply test/jdk/java/nio/channels/Selector/WakeupNow.java:66]: Hi @jaikiran . Yes it was unused code intentionally removed.

**Comment by bplb** [comment]
> Changing it to use nanoTime is good but I wonder about the reliability of the select(2000) taking >= 1s.

I am not sure about it either, but I thought that with the addition of the time delta to the exception message it might allow some information gathering about failures should they occur.

**Comment by AlanBateman** [comment]
> > Changing it to use nanoTime is good but I wonder about the reliability of the select(2000) taking >= 1s.
> 
> I am not sure about it either, but I thought that with the addition of the time delta to the exception message it might allow some information gathering about failures should they occur.

I don't know which clock source is used by GetQueuedCompletionStatus but I agree, using nanoTimes would be better here and if it fails because it takes <1s then it at least tells us something.

**Review by AlanBateman — APPROVED** [review]

**Pull request closed without merging**